src/reminder/reminder.service.ts

```diff
@@ -388,27 +414,58 @@ export class ReminderService {
     };
   }
 
-  async markAsNotComplete(rid: number): Promise<string> {
-    const reminder = await this.reminderRepository.findOne({ where: { rid: rid } });
+  async markAsNotComplete(rid: number, uid: number): Promise<string> {
+    const reminder = await this.reminderRepository.findOne({ where: { rid: rid, uid }, relations: ['recurring'] });
     if (!reminder) throw new HttpException('Reminder not found', HttpStatus.NOT_FOUND);
     if (!reminder.isDone) throw new HttpException('This reminder is not yet completed', HttpStatus.CONFLICT);
     reminder.isDone = false;
-
+    const schedule = this.getScheduleFromRecurring(reminder.rid, reminder.recurrings, reminder.startingDateTime);
+    await this.scheduleReminder(reminder, schedule);
     await this.reminderRepository.save(reminder);
     return 'Complete';
   }
 
-  async markAsComplete(rid: number, currentDate: Date): Promise<string> {
-    const reminder = await this.reminderRepository.findOne({ where: { rid: rid } });
+  async markAsComplete(rid: number, currentDate: Date, uid: number): Promise<string> {
+    const reminder = await this.reminderRepository.findOne({ where: { rid: rid, uid } });
     if (!reminder) throw new HttpException('Reminder not found', HttpStatus.NOT_FOUND);
     if (reminder.isDone) throw new HttpException('This reminder is already completed', HttpStatus.CONFLICT);
     if (reminder.startingDateTime.getTime() - currentDate.getTime() > 1800000)
       throw new HttpException('Can not mark reminder in advance over 30 minutes', HttpStatus.CONFLICT);
     if (reminder.recurrings.length !== 0) throw new HttpException('Can not mark reminder that have recurring', HttpStatus.CONFLICT);
     // not sure about reminder.recurrings.length
     reminder.isDone = true;
-
     await this.reminderRepository.save(reminder);
+    this.clearReminderJobs(reminder.rid);
     return 'Complete';
   }
+  private clearReminderJobs(rid: number): void {
+    this.schedulerService.deleteJob(rid.toString(), JobType.INTERVAL);
```

Comment:
same as above

Reply:
all services in scheduler do not return promise